xchange-gateio/src/main/java/org/knowm/xchange/gateio/GateioAdapters.java

```diff
@@ -237,11 +242,26 @@ public static ExchangeMetaData adaptToExchangeMetaData(
               btermarketInfo.getDecimalPlaces(),
               null);
       currencyPairs.put(currencyPair, currencyPairMetaData);
+      GateioCoin gateioCoin = coins.get(currencyPair.base.getCurrencyCode());
+      GateioFeeInfo gateioFeeInfo = gateioFees.get(currencyPair.base.getCurrencyCode());
+      if(gateioCoin != null && gateioFeeInfo != null) {
+        currencies.put(currencyPair.base, adaptCurrencyMetaData(gateioCoin, gateioFeeInfo));
```

Comment:
Currencies metadata can be converted in a separate loop for more robust and convenient conversion logic